services/mailer/mailer.go

```diff
@@ -147,42 +147,54 @@ type smtpSender struct{}
 func (s *smtpSender) Send(from string, to []string, msg io.WriterTo) error {
 	opts := setting.MailService
 
-	host, port, err := net.SplitHostPort(opts.Host)
-	if err != nil {
-		return err
+	var network string
+	var address string
+	if opts.Protocol == "smtp+unix" {
+		network = "unix"
+		address = opts.SmtpAddr
+	} else {
+		network = "tcp"
+		address = net.JoinHostPort(opts.SmtpAddr, opts.SmtpPort)
 	}
 
-	tlsconfig := &tls.Config{
-		InsecureSkipVerify: opts.SkipVerify,
-		ServerName:         host,
+	conn, err := net.Dial(network, address)
+	if err != nil {
+		return err
 	}
+	defer conn.Close()
 
-	if opts.UseCertificate {
-		cert, err := tls.LoadX509KeyPair(opts.CertFile, opts.KeyFile)
-		if err != nil {
-			return err
+	var tlsconfig *tls.Config
+	if opts.Protocol == "smtps" || opts.Protocol == "smtp+startls" {
+		tlsconfig = &tls.Config{
+			InsecureSkipVerify: opts.ForceTrustServerCert,
+			ServerName:         opts.SmtpAddr,
 		}
-		tlsconfig.Certificates = []tls.Certificate{cert}
-	}
 
-	conn, err := net.Dial("tcp", net.JoinHostPort(host, port))
-	if err != nil {
-		return err
+		if opts.UseClientCert {
+			cert, err := tls.LoadX509KeyPair(opts.ClientCertFile, opts.ClientKeyFile)
+			if err != nil {
+				return err
```

Comment:
```suggestion
				return fmt.Errorf("could not load provided client certificates: %v", err)
```

Reply:
I'll update all the errors to do that; the old code just did `return err` which is why I copied it.